cypress-io/cypress
# fix: unify cdp approach to fix devtools in electron

Closes #15932

### Additional details


Electron has [issues](https://github.com/electron/electron/issues/34260) when you attach to `win.webContents.debugger` where it causes devtools to get out of sync with the DOM. This can be fixed by attaching to electron in a similar fashion to how we attach to other chromium browsers. This PR makes that transition

### Steps to test


* Ensure electron tests are passing
* Use the DOM inspector in electron and ensure that the DOM is showing up correctly

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
